train_segmentation_model.py

```diff
@@ -89,17 +89,50 @@ def train(gcp_bucket, config_file, random_module_global_seed, numpy_random_globa
     validation_generator = ImagesAndMasksGenerator(
         Path(local_dataset_dir, train_config['dataset_id'],
              'validation').as_posix(),
-        rescale=1./255,
+        rescale=1. / 255,
         target_size=target_size,
         batch_size=batch_size,
         seed=None if 'validation_data_shuffle_seed' not in train_config else train_config['validation_data_shuffle_seed'])
 
+    if pretrained_model_id is not None:
+        local_pretrained_model_dir = Path(tmp_directory, 'pretrained_models')
+        copy_folder_locally_if_missing(os.path.join(gcp_bucket, 'models', pretrained_model_id), local_pretrained_model_dir)
+        path_pretrained_model = Path(local_pretrained_model_dir, pretrained_model_id, "model.hdf5").as_posix()
+
+        with Path(local_pretrained_model_dir, pretrained_model_id, 'config.yaml').open('r') as f:
+            pretrained_model_config = yaml.safe_load(f)['train_config']
+
+        with Path(local_pretrained_model_dir, pretrained_model_id, 'metadata.yaml').open('r') as f:
+            pretrained_model_metadata = yaml.safe_load(f)
+
+        pretrained_info = {'pretrained_model_id': pretrained_model_id, 'pretrained_config': pretrained_model_config, 'pretrained_metadata': pretrained_model_metadata}
+
+        # confirm that the same model and backbone are used from pretrain and train
+        print(pretrained_model_config['segmentation_model']['model_name'], train_config['segmentation_model']['model_name'])
+        assert pretrained_model_config['segmentation_model']['model_name'] == train_config['segmentation_model']['model_name']
+        print(pretrained_model_config['segmentation_model']['model_parameters']['backbone_name'], train_config['segmentation_model']['model_parameters']['backbone_name'])
+        assert pretrained_model_config['segmentation_model']['model_parameters']['backbone_name'] == train_config['segmentation_model']['model_parameters']['backbone_name']
+        # same loss function
```

Comment:
this might be another instance to revisit. again, easiest way for now is actually just assert it like u r (ignore my prior comment). the real issue comes down to asserting binary loss functions (sigmoid mapped predictions) vs. categorical loss functions (softmax mapped predictions) rather than any one specific function. for example, there are variants of cross entropy, dice loss, jaccard loss, etc. currently we don't allow activation selection (sigmoid vs softmax) in train config, tho @BrendenBarbour may add this in his alt. models PR. here is an example of where activation is defaulted to currently (https://github.com/qubvel/segmentation_models/blob/master/segmentation_models/models/unet.py#L116).

long story short, leave loss fcn asserted, and we'll revisit with more nuance in brenden's PR i think